### ISSUE TITLE
Zarr version 2 or 3

### ISSUE THREAD

**Issue by normanrz** [abcucberkeley/c-zarr#8]
Thanks for making this library available. I think it is nice to have an alternative in C/C++ besides tensorstore.

Looking through the code, I am a bit confused whether this library supports Zarr versions [2](https://zarr-specs.readthedocs.io/en/latest/v2/v2.0.html) or [3](https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html) or both?
Here, I can see that the `"zarr_format": 2` is used: https://github.com/abcucberkeley/cpp-zarr/blob/main/src/zarr.cpp#L29
And here, I the sharding codec is checked, which is only available in version 3: https://github.com/abcucberkeley/cpp-zarr/blob/main/src/zarr.cpp#L43-L46
However, for version 3 other metadata keys have changed (e.g. `dimension_separator` -> `chunk_key_encoding`) and the json files are now all called `zarr.json` instead of `.zarray`.

**Comment by MattHazard** [comment]
Thank you for your interest in cpp-zarr. We currently only support v2 of the Zarr spec but we were experimenting with adding v3 features. We added sharding first because we had use cases where we thought sharding could provide major speed increases for us.

**Comment by normanrz** [comment]
Thanks for the reply! We might have some capacity to contribute to your library in the coming months. Would you be interested in PRs that improve support for Zarr v3?

**Comment by MattHazard** [comment]
Yes, we would be interested in that. I think having some real support for Zarr v3 would be great for the library.

**Comment by MattHazard** [comment]
I'm going to close this issue for now. Feel free to open another issue if there is ever any renewed interest in the future.
